File: src/main/java/gg/climb/lolobjects/game/Champion.java
package gg.climb.lolobjects.game;

import gg.climb.lolobjects.InternalId;
import gg.climb.lolobjects.RiotId;

import org.immutables.value.Value;

/**
 * A class that represents a champion in League. Obtain an instance via {@link ChampionBuilder}.
 */
@Value.Style(visibility = Value.Style.ImplementationVisibility.PRIVATE,
             strictBuilder = true)
@Value.Immutable
public abstract class Champion {
  public abstract InternalId getInternalId();

  public abstract String getName();

  public abstract RiotId getRiotId();

  public abstract ChampionStats getStats();

  public abstract ChampionImage getImage();
}

File: src/main/scala/gg/climb/lolobjects/game/Champion.scala
package gg.climb.lolobjects.game

import gg.climb.lolobjects.{InternalId, RiotId}

/**
 * A class that represents a champion in League.
 */
class Champion(val internalId: InternalId[Champion],
               val riotId: RiotId[Champion],
               val name: String,
               val stats: ChampionStats,
               val image: ChampionImage){

  override def toString = s"$internalId=$internalId, riotId=$riotId, name=$name, stats=$stats, image=$image"
}

object Champion{
  def apply( internalId: InternalId[Champion],
             riotId: RiotId[Champion],
             name: String,
             stats: ChampionStats,
             image: ChampionImage) = new Champion(internalId, riotId, name, stats, image)
}
